src/App.tsx

```diff
@@ -1,19 +1,208 @@
+  titleError: boolean;
+  nameError: boolean;
+  maxTitleLength: number;
 };
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    title: '',
+    name: '',
+    addedTodos: [...preparedTodos],
+    titleError: false,
+    nameError: false,
+    maxTitleLength: 20,
+  };
+
+  handleChange = (event: {
+    target: {
+      name: string;
+      value: string;
+    };
+  }): void => {
+    const {
+      name,
+      value,
+    } = event.target;
+
+    this.setState(state => ({
+      ...state,
+      [name]: value,
+    }));
+  };
+
+  handleFocus = () => {
+    const { titleError, nameError } = this.state;
+
+    if (titleError) {
+      this.setState(state => ({
+        titleError: !state.titleError,
+      }));
+    }
+
+    if (nameError) {
+      this.setState(state => ({
+        nameError: !state.nameError,
+      }));
+    }
+  };
+
+  addTodo = () => {
+    if (!this.state.title.trim().length) {
+      this.setState(state => ({
+        titleError: !state.titleError,
+      }));
+
+      return;
+    }
+
+    if (this.state.title.match(/[^a-zA-Zа-яА-Я0-9]/)) {
+      this.setState(state => ({
+        titleError: !state.titleError,
+      }));
+
+      return;
+    }
+
+    if (!this.state.name.length) {
+      this.setState(state => ({
+        nameError: !state.nameError,
+      }));
+
+      return;
+    }
+
+    const todoId = () => {
+      if (this.state.addedTodos.length === 0) {
+        return 1;
+      }
+
+      const array = this.state.addedTodos.map(item => item.id);
+      const maxNum = Math.max(...array);
+
+      return maxNum + 1;
+    };
+
+    const addedUser = users.find(user => user.name === this.state.name);
+    const newTodo = {
+      userId: addedUser?.id,
+      id: todoId(),
+      title: this.state.title.trim().slice(0, this.state.maxTitleLength),
+      completed: false,
+      user: addedUser || null,
+    };
+
+    this.setState(state => ({
+      addedTodos: [...state.addedTodos, newTodo],
+      title: '',
+      name: '',
+    }));
+  };
+
+  removeTodo = (id: number) => {
+    this.setState(state => ({
+      addedTodos: state.addedTodos.filter(item => item.id !== id),
+    }));
+  };
+
+  switchTodo = (id: number, completed: boolean) => {
+    this.setState(state => ({
+      addedTodos: state.addedTodos.map(todo => {
+        if (todo.id !== id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          completed: !completed,
+        };
+      }),
+    }));
+  };
+
+  render() {
+    const { addedTodos } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Todos list</h1>
+
+        <TodoList
+          preparedTodos={addedTodos}
+          removeTodo={this.removeTodo}
+          switchTodo={this.switchTodo}
+        />
+
+        <h2>Here you can add you todo to the list</h2>
+
+        <form onSubmit={(event) => {
+          event.preventDefault();
+          this.addTodo();
+        }}
```

Comment:
move event.preventDefault() inside addTodo

```suggestion
        <form onSubmit={this.addTodo}
```